### DISCUSSION TITLE
Next.js API route error on Vercel - Unhandled Promise Rejection

### DISCUSSION THREAD

**Discussion by csvidit** [vercel/next.js#52755] [Help] [answered]
### Summary

The code works perfectly fine on my local machine using a development server, but on the deployed version on Vercel, the API route call always fails. For reference, I am on an M1 MacBook Air, 16 GB unified memory. 

This is the code for the project - https://github.com/csvidit/turbobiz

This is the link to the production build, hosted on Vercel - https://turbobiz.viditkhandelwal.com

This is the error I get in the Serverless Function logs on Vercel -

Unhandled Promise Rejection 	{"errorType":"Runtime.UnhandledPromiseRejection","errorMessage":"Error: EMFILE: too many open files, open '/var/task/node_modules/@firebase/auth/dist/node-esm/index.js'","reason":{"errorType":"Error","errorMessage":"EMFILE: too many open files, open '/var/task/node_modules/@firebase/auth/dist/node-esm/index.js'","code":"EMFILE","errno":-24,"syscall":"open","path":"/var/task/node_modules/@firebase/auth/dist/node-esm/index.js","stack":["Error: EMFILE: too many open files, open '/var/task/node_modules/@firebase/auth/dist/node-esm/index.js'","    at Object.openSync (node:fs:601:3)","    at Object.readFileSync (node:fs:469:35)","    at Module._extensions..js (node:internal/modules/cjs/loader:1271:18)","    at Module.load (node:internal/modules/cjs/loader:1117:32)","    at Module._load (node:internal/modules/cjs/loader:958:12)","    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:169:29)","    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)"]},"promise":{},"stack":["Runtime.UnhandledPromiseRejection: Error: EMFILE: too many open files, open '/var/task/node_modules/@firebase/auth/dist/node-esm/index.js'","    at process.<anonymous> (file:///var/runtime/index.mjs:1189:17)","    at process.emit (node:events:525:35)","    at emit (node:internal/process/promises:149:20)","    at processPromiseRejections (node:internal/process/promises:283:27)","    at process.processTicksAndRejections (node:internal/process/task_queues:96:32)"]}
Unknown application error occurred
Runtime.Unknown

I have spent 4 hours searching for fixes, but nothing has worked yet. Let me know if you need any more information or context.

**Comment by icyJoseph** [comment, marked as answer]
Mmm looks like it is something with phosphor icons.

- https://github.com/phosphor-icons/react/issues/45#issuecomment-1488729840
- https://github.com/vercel/next.js/issues/52646

**Comment by csvidit** [comment]
That solved it! Looking at the errors I thought it was related to Firebase Auth but nice catch.
